### PR TITLE
[tests] Mark tests where they should be run in

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: hail/python/test/hail/helpers.py [hunk 1]
```diff
@@ -121,6 +121,16 @@ def create_all_values_datasets():
     return (create_all_values_table(), create_all_values_matrix_table())
 
 
+def run_in(*locations, clouds=None):
+    for location in locations:
+        if location not in ('local', 'spark', 'batch', 'all'):
+            raise RuntimeError(f'unknown location {location}')
```
Comment (danking) [inline hail/python/test/hail/helpers.py:127]: let's error if 'all' is specified with any other location.

File: hail/python/test/hail/methods/relatedness/test_pc_relate.py [hunk 2]
```diff
@@ -51,7 +53,7 @@ def test_pc_relate_simple_example():
 
 
 @test_timeout(6 * 60, batch=14 * 60)
-@skip_when_service_backend_in_azure(reason='takes >14 minutes in QoB in Azure')
+@run_in('local', 'spark', 'batch', clouds=['gcp'])  # reason='takes >14 minutes in QoB in Azure'
```
Comment (danking) [inline hail/python/test/hail/methods/relatedness/test_pc_relate.py:56]: should be 'all' right?

File: hail/python/test/hail/helpers.py (hunk 1 above)
Comment (danking) [reply hail/python/test/hail/helpers.py:127]: Maybe also error if all three are specified? Best to enforce a consistent, simple style across everything.

**Review by danking — CHANGES_REQUESTED** [review]

**Review by danking — CHANGES_REQUESTED** [review]
I reviewed by hand every single instance where run_in was not 'all' and some cases where it was 'all'. These are my only comments. I think we should work hard to merge this soon, it will reduce the churn in CI and increase developer velocity, the most important thing we have.

File: hail/python/test/hail/expr/test_expr.py [hunk 3]
```diff
@@ -781,6 +835,7 @@ def test_agg_explode(self):
         for aggregation, expected in tests:
             self.assertEqual(t.aggregate(aggregation), expected)
 
+    @run_in('local')
```
[the remark this answers is not in the document]
Comment (patrick-schultz) [reply hail/python/test/hail/expr/test_expr.py:838]: Agreed. We were too quick to say everything in test_expr was just local computation.

File: hail/python/test/hail/expr/test_expr.py [hunk 4]
```diff
@@ -4158,11 +4412,13 @@ def test_reservoir_sampling():
         assert abs(mean - sample_mean) / expected_stdev < 4 , (iteration, sample_size, abs(mean - sample_mean) / expected_stdev)
 
 
+@run_in('local')
 def test_local_agg():
```
[the remark this answers is not in the document]
Comment (danking) [reply hail/python/test/hail/expr/test_expr.py:4416]: 👍 misread this one. you're right.

File: hail/python/test/hail/methods/test_vep.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
+import unittest
```
[the remark this answers is not in the document]
Comment (jigold) [reply hail/python/test/hail/methods/test_vep.py:1]: reverted that change. I forgot why I did it other than at the time all of the other tests were meant to be local, but now they're not.

**Comment by jigold** [comment]
Should be good to go. Please double check I got all of the changes you wanted.

**Comment by jigold** [comment]
Closing for now.

**Comment by jigold** [comment]
Closing again for now.

**Pull request closed without merging**